src/main/java/util/XmlParser.java

```diff
@@ -38,76 +58,55 @@ private XmlParser convertFileToDomDocument(String pathToRawXml) {
         try {
             DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
             DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
-            document = dBuilder.parse(rawXmlFile);
-            document.getDocumentElement().normalize();
+            domDocument = dBuilder.parse(rawXmlFile);
+            domDocument.getDocumentElement().normalize();
         } catch (ParserConfigurationException | IOException | SAXException e) {
             e.printStackTrace();
         }
         return this;
     }
 
+    private XmlParser convertDomDocumentToString() {
+        try {
+            DOMSource domSource = new DOMSource(domDocument);
+            StringWriter writer = new StringWriter();
+            StreamResult result = new StreamResult(writer);
+            TransformerFactory tf = TransformerFactory.newInstance();
+            Transformer transformer = tf.newTransformer();
+            transformer.transform(domSource, result);
+            String xmlDocumentAsString = writer.toString().replaceAll("\n|\r", "");
```

Comment:
It seems you wanted to save result to class variable. Here you write to local variable.